src/Utils/system_utils.py

```diff
@@ -0,0 +1,20 @@
+import sys
+import re
+import torch
+
+
+def get_device():
+    if sys.platform == "darwin":
+        return "mps" if torch.backends.mps.is_available() else "cpu"
+
+    return "cuda" if torch.cuda.is_available() else "cpu"
+
+def extract_file_path(input_str):
```

Comment:
By looking at this method, it's not clear the purpose of it. Can you rename it and also consider running on windows environment? Because the file path separator is different.